query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/converters/PinotFunctionConverterConfig.java

```diff
@@ -1,22 +1,30 @@
 package org.hypertrace.core.query.service.pinot.converters;
 
+import com.google.common.collect.ImmutableSet;
 import com.typesafe.config.Config;
 import com.typesafe.config.ConfigFactory;
+import java.util.Collections;
+import java.util.Set;
+import javax.annotation.Nonnull;
+import lombok.AccessLevel;
 import lombok.AllArgsConstructor;
 import lombok.Value;
 
 @Value
-@AllArgsConstructor
+@AllArgsConstructor(access = AccessLevel.PACKAGE)
 public class PinotFunctionConverterConfig {
 
   private static final String PERCENTILE_AGGREGATION_FUNCTION_CONFIG = "percentileAggFunction";
   private static final String DISTINCT_COUNT_AGGREGATION_FUNCTION_CONFIG =
       "distinctCountAggFunction";
+  private static final String ARGS_FOR_ACCURATE_DISTINCT_COUNT_AGG_CONFIG =
```

Comment:
This isn't super generic, since the distinct count function it self is configurable. Can we update the config so we basically have a default distinct count an an override map? (either function to list of column names or column name to function)

Reply:
done